cmd/agent/api/agent/agent.go

```diff
@@ -80,13 +81,23 @@ func getHostname(w http.ResponseWriter, r *http.Request) {
 }
 
 func makeFlare(w http.ResponseWriter, r *http.Request) {
+	var profile *flare.Profile
+
+	err := json.NewDecoder(r.Body).Decode(&profile)
+	switch {
+	case err == io.EOF:
+		log.Debugf("No profile, continuing.")
+	case err != nil:
+		log.Errorf("Error while decoding profile from request body: %s", err)
```

Comment:
We should return an error here instead of sending the flare.

Reply:
You mean in the second case?

Reply:
yes, if the json decode fails